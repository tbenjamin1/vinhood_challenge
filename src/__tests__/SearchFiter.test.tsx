import React from "react";
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import {render,screen} from '@testing-library/react'
import { fetchBreweries } from '@redux-slices/breweries.slice'
const mockStore = configureStore([]);

import SearchFilter from "../components/searchFilter/index";

it('render filter page',() => {
 let  store = mockStore({
      myState: 'filter test',
      breweries: {
        breweries: [],
        status: 'idle',
        error: null,
      },
    });
   render(<Provider store={store}><SearchFilter/> </Provider>);
   expect(screen.getByTestId('filter'))
})

describe('My Connected React-Redux Component', () => {
    let store:any;
    let component:any;
  
    beforeEach(() => {
      store = mockStore({
        myState: 'sample text',
        breweries: {
            breweries: [],
            status: 'idle',
            error: null,
          },
      });
  
      component = render(
        <Provider store={store}>
          <SearchFilter />
        </Provider>
      );
    });
  
    it('should render with given state from Redux store', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });

    it('should dispatch an action on button click', () => {
        renderer.act(() => {
          component.root.findByType('button').props.onClick();
        });
        renderer.act(() => {
            component.root.findByType('input')
              .props.onChange({ target: { value: 'satte name' } });
          });
    
        expect(store.dispatch).toHaveBeenCalledTimes(1);
        expect(store.dispatch).toHaveBeenCalledWith(
            fetchBreweries({ city:'kigali',
                type:'micro',
                name:'brewery',
                sort: 'desc',
                perPage:6, })
        );
      });

 
  
    
  });
 




  