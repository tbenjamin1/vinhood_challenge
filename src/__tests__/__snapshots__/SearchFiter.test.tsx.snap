// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="filter"
      >
        <form
          class="py-4"
        >
          <div
            class="collapse collapse-plus border border-gray-100 bg-gray-200"
          >
            <input
              type="checkbox"
            />
            <div
              class="collapse-title  text-lg font-medium"
            >
              Filter By
            </div>
            <div
              class="collapse-content bg-white"
            >
              <div
                class="form-control my-3"
              >
                <label
                  class="label"
                  date-testid="home-page"
                  for="city"
                >
                  Name
                </label>
                <input
                  class="input input-bordered"
                  placeholder="eg: Camilia"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-control my-3"
              >
                <label
                  class="label"
                  for="city"
                >
                  City
                </label>
                <input
                  class="input input-bordered"
                  placeholder="eg: New York"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="form-control my-3"
              >
                <label
                  class="label"
                  for="type"
                >
                  Type
                </label>
                <select
                  class="select select-bordered capitalize w-full"
                  id="type"
                >
                  <option
                    value="micro"
                  >
                    micro
                  </option>
                  <option
                    value="nano"
                  >
                    nano
                  </option>
                  <option
                    value="regional"
                  >
                    regional
                  </option>
                  <option
                    value="brewpub"
                  >
                    brewpub
                  </option>
                  <option
                    value="large"
                  >
                    large
                  </option>
                  <option
                    value="planning"
                  >
                    planning
                  </option>
                  <option
                    value="bar"
                  >
                    bar
                  </option>
                  <option
                    value="contract"
                  >
                    contract
                  </option>
                  <option
                    value="proprietor"
                  >
                    proprietor
                  </option>
                  <option
                    value="closed"
                  >
                    closed
                  </option>
                </select>
              </div>
              <div
                class="form-control my-3"
              >
                <label
                  class="label justify-start gap-2"
                  for="sort"
                >
                  <span>
                    Desc
                  </span>
                  <input
                    class="toggle"
                    id="sort"
                    type="checkbox"
                    value="0"
                  />
                  <span>
                    Asc
                  </span>
                </label>
              </div>
            </div>
          </div>
          <div
            class="collapse collapse-plus border border-gray-100 bg-gray-200"
          >
            <input
              type="checkbox"
            />
            <div
              class="collapse-title  text-lg font-medium"
            >
              Per page
            </div>
            <div
              class="collapse-content bg-white"
            >
              <label
                class="label"
              >
                Per Page: 
                5
              </label>
              <input
                class="range"
                max="100"
                min="0"
                step="5"
                type="range"
                value="5"
              />
            </div>
          </div>
          <div
            class="my-3 flex justify-end gap-2"
          >
            <button
              class="btn"
              data-testid="filter-btn"
            >
              cancel
            </button>
            <button
              class="btn  btn-primary"
              style="background-color: blue;"
            >
              Apply 
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="filter"
    >
      <form
        class="py-4"
      >
        <div
          class="collapse collapse-plus border border-gray-100 bg-gray-200"
        >
          <input
            type="checkbox"
          />
          <div
            class="collapse-title  text-lg font-medium"
          >
            Filter By
          </div>
          <div
            class="collapse-content bg-white"
          >
            <div
              class="form-control my-3"
            >
              <label
                class="label"
                date-testid="home-page"
                for="city"
              >
                Name
              </label>
              <input
                class="input input-bordered"
                placeholder="eg: Camilia"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-control my-3"
            >
              <label
                class="label"
                for="city"
              >
                City
              </label>
              <input
                class="input input-bordered"
                placeholder="eg: New York"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-control my-3"
            >
              <label
                class="label"
                for="type"
              >
                Type
              </label>
              <select
                class="select select-bordered capitalize w-full"
                id="type"
              >
                <option
                  value="micro"
                >
                  micro
                </option>
                <option
                  value="nano"
                >
                  nano
                </option>
                <option
                  value="regional"
                >
                  regional
                </option>
                <option
                  value="brewpub"
                >
                  brewpub
                </option>
                <option
                  value="large"
                >
                  large
                </option>
                <option
                  value="planning"
                >
                  planning
                </option>
                <option
                  value="bar"
                >
                  bar
                </option>
                <option
                  value="contract"
                >
                  contract
                </option>
                <option
                  value="proprietor"
                >
                  proprietor
                </option>
                <option
                  value="closed"
                >
                  closed
                </option>
              </select>
            </div>
            <div
              class="form-control my-3"
            >
              <label
                class="label justify-start gap-2"
                for="sort"
              >
                <span>
                  Desc
                </span>
                <input
                  class="toggle"
                  id="sort"
                  type="checkbox"
                  value="0"
                />
                <span>
                  Asc
                </span>
              </label>
            </div>
          </div>
        </div>
        <div
          class="collapse collapse-plus border border-gray-100 bg-gray-200"
        >
          <input
            type="checkbox"
          />
          <div
            class="collapse-title  text-lg font-medium"
          >
            Per page
          </div>
          <div
            class="collapse-content bg-white"
          >
            <label
              class="label"
            >
              Per Page: 
              5
            </label>
            <input
              class="range"
              max="100"
              min="0"
              step="5"
              type="range"
              value="5"
            />
          </div>
        </div>
        <div
          class="my-3 flex justify-end gap-2"
        >
          <button
            class="btn"
            data-testid="filter-btn"
          >
            cancel
          </button>
          <button
            class="btn  btn-primary"
            style="background-color: blue;"
          >
            Apply 
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
